pkg/plugin/aggregation/run.go

```diff
@@ -209,3 +203,57 @@ func Cleanup(client kubernetes.Interface, plugins []plugin.Interface) {
 		p.Cleanup(client)
 	}
 }
+
+// RunAndMonitorPlugin will start a plugin then monitor it for errors starting/running.
+// Errors detected will be handled by saving an error result in the aggregator.Results.
+func (a *Aggregator) RunAndMonitorPlugin(ctx context.Context, p plugin.Interface, client kubernetes.Interface, nodes []corev1.Node, address string, cert *tls.Certificate) {
+	monitorCh := make(chan *plugin.Result, 1)
+	pCtx, cancel := context.WithCancel(ctx)
+
+	if err := p.Run(client, address, cert); err != nil {
+		err := errors.Wrapf(err, "error running plugin %v", p.GetName())
+		logrus.Error(err)
+		monitorCh <- utils.MakeErrorResult(p.GetResultType(), map[string]interface{}{"error": err.Error()}, "")
+	}
+
+	go p.Monitor(pCtx, client, nodes, monitorCh)
+	go a.IngestResults(pCtx, monitorCh)
+
+	// Control loop; check regularly if we have results or not for this plugin. If results are in,
+	// then stop the go routines monitoring the plugin. If the parent context is cancelled, stop monitoring.
+	for {
+		select {
+		case <-ctx.Done():
+			cancel()
+			return
+		case <-after(10 * time.Second):
+		default:
```

Comment:
I think having the `default` case in here will result in the "sleep" not taking place. On each iteration, the default case will be satisfied and it will fall through to performing the operations after the `select`. This applies to all the other changes in sleep logic below.

Reply:
Yes thank you for catching this!

Reply:
After removing that I did find that my test was failing because of how each package was using their own, unexported, `after` method. So that meant when doing this `RunAndMonitor` the `job.Monitor` logic was using its own method which couldn't be edited because it was unexported. As a result _it_ was sleeping for 10s before polling on the status.

The solution I went with was to centralize the `after` into a little `pkg/time` package and added another `pkg/time/timetest` (ala httptest from standard repo) to provide some helpful functions to shorten/ignore/reset the method.

Tests back to passing.